src/gasPriceOracle/oracle.ts

```diff
@@ -1,51 +1,79 @@
 import assert from "assert";
-import { Transport } from "viem";
-import { providers } from "ethers";
+import { PublicClient, Transport } from "viem";
+import { PopulatedTransaction, providers } from "ethers";
 import { CHAIN_IDs } from "../constants";
 import { BigNumber, chainIsOPStack } from "../utils";
-import { GasPriceEstimate } from "./types";
+import { GasPriceEstimate, InternalGasPriceEstimate } from "./types";
 import { getPublicClient } from "./util";
 import * as arbitrum from "./adapters/arbitrum";
 import * as ethereum from "./adapters/ethereum";
 import * as linea from "./adapters/linea";
 import * as polygon from "./adapters/polygon";
-import * as arbitrumViem from "./adapters/arbitrum-viem";
 import * as lineaViem from "./adapters/linea-viem";
-import * as polygonViem from "./adapters/polygon-viem";
+import { PolygonGasStation } from "./adapters/polygon";
+
+export interface GasPriceEstimateOptions {
+  // baseFeeMultiplier Multiplier applied to base fee for EIP1559 gas prices (or total fee for legacy).
+  baseFeeMultiplier: number;
+  // legacyFallback In the case of an unrecognized chain, fall back to type 0 gas estimation.
+  legacyFallback: boolean;
+  // chainId The chain ID to query for gas prices. If omitted can be inferred by provider.
+  chainId: number;
+  // unsignedTx The unsigned transaction used for simulation by Linea's Viem provider to produce the priority gas fee.
+  unsignedTx?: PopulatedTransaction;
+  // transport Viem Transport object to use for querying gas fees used for testing.
+  transport?: Transport;
+  // polygonGasStation Custom Polygon GasStation class used for testing.
+  polygonGasStation?: PolygonGasStation;
```

Comment:
I'm not super keen on exposing this type in the general options type. It's intended to be hidden behind the Polygon adapter and is really an internal type. Can we maintain that while still permitting this to be testable?

Reply:
I agree with that, best I could think of is using an env var to toggle it: https://github.com/across-protocol/sdk/pull/801/commits/576dac68ab79f46b6e12d7ac1e1404b79390a522

Reply:
I think I have a solution but it's low priority - let's just get this in as-is to unblock the API.